### PR TITLE
 NGINXWebServer

### PR DESCRIPTION
**Pull request by yossifirouz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/NGINXWebServer/ModelingRules/NGINXWebServer/NGINXWebServerModelingRules.xif [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+| alter XDM.Network.event_timestamp= request_time
+| alter XDM.Network.Source.user.username =username
+| alter XDM.Network.http.url = requested_line
+| alter XDM.Network.http.response_code = response_code  
+| alter  XDM.Network.http.user_agent= User_agent
+| alter XDM.Network.http.referrer =Referrer
+| alter  XDM.Network.Destination.bytes=bytes_size
+| alter XDM.Network.Source.ipv4 =source_ip
+| alter XDM.Network.Source.user.username =username
```
Comment (evisochek) [inline Packs/NGINXWebServer/ModelingRules/NGINXWebServer/NGINXWebServerModelingRules.xif:29]: missing ; in the end

File: Packs/NGINXWebServer/ModelingRules/NGINXWebServer/NGINXWebServerModelingRules.xif [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+| alter Referrer= arrayindex(regextract(_raw_log ,"(http[^\"]+)"),0)
+| alter bytes_size=arrayindex(regextract(_raw_log ,"\d\s(\d+)"),0)
+// extract source_ip
+| alter sourceipv4= arrayindex(regextract(_raw_log , "(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\s)") ,0)
+| alter sourceipv6 = arrayindex(regextract(_raw_log ,"(\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+)") ,0)
+| alter sourceip = coalesce(sourceipv4 ,sourceipv6)
+| fields -sourceipv4, sourceipv6
+// end extract source_ip
+| alter Username = arrayindex(regextract(_raw_log,"(\S+)\s\[\d+\/") ,0)
+
```
Comment (evisochek) [inline Packs/NGINXWebServer/ModelingRules/NGINXWebServer/NGINXWebServerModelingRules.xif:19]: extra whitespace

File: Packs/NGINXWebServer/ModelingRules/NGINXWebServer/NGINXWebServerModelingRules.xif [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+| alter XDM.Audit.threat.severity = log_level
+| alter XDM.Audit.identity.uuid=tid 
+| alter XDM.Network.Source.ipv4 =client_address       
+| alter XDM.Audit.threat.description = error_message
+| alter XDM.Network.Destination.host.ipv4_addresses= server
+| alter XDM.Network.http.url=request 
+| alter XDM.Network.Observer.unique_identifier =host
+| alter XDM.Network.http.referrer= referrer 
+| alter XDM.Audit.TriggeredBy.identity.uuid =  uri
+| alter XDM.Network.Destination.port = destination_port
```
Comment (evisochek) [inline Packs/NGINXWebServer/ModelingRules/NGINXWebServer/NGINXWebServerModelingRules.xif:61]: missing ; in the end

File: Packs/NGINXWebServer/ModelingRules/NGINXWebServerModelingRules.xif [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+| alter User_agent = arrayindex(regextract(_raw_log ,"\"\s\"([^\"]+)") ,0)
+| alter Referrer= arrayindex(regextract(_raw_log ,"(http[^\"]+)"),0)
+| alter bytes_size=arrayindex(regextract(_raw_log ,"\d\s(\d+)"),0)
+// extract source_ip
+| alter sourceipv4= arrayindex(regextract(_raw_log , "(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\s)") ,0)
+| alter sourceipv6 = arrayindex(regextract(_raw_log ,"(\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+)") ,0)
+| alter sourceip = coalesce(sourceipv4 ,sourceipv6)
+// end extract source_ip
+| alter Username = arrayindex(regextract(_raw_log,"(\S+)\s\[\d+\/") ,0)
+| alter XDM.Network.event_timestamp= parse_timestamp("%F %X", Request_time)
```
Comment (guytamir10) [inline Packs/NGINXWebServer/ModelingRules/NGINXWebServerModelingRules.xif:16]: need to replace the parse_timestamp to:

%d/%b/%Y:%H:%M:%S %Ez

File: Packs/NGINXWebServer/ModelingRules/NGINXWebServerModelingRules.xif [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+| alter XDM.Network.http.url = requested_line
+| alter XDM.Network.http.response_code = response_code  
+//| alter XDM.Network.http.user_agent= User_agent
+| alter XDM.Network.http.referrer =Referrer
+| alter XDM.Network.Destination.bytes=to_number(bytes_size)
+| alter XDM.Network.Source.ipv4 =source_ip
+| alter XDM.Network.Source.user.username =Username;
+
+[MODEL: dataset="nginx_nginx_raw", model="Audit"]
```
Comment (guytamir10) [inline Packs/NGINXWebServer/ModelingRules/NGINXWebServerModelingRules.xif:25]: from where you extract the data? nothing from the parser is not working.

File: Packs/NGINXWebServer/ModelingRules/NGINXWebServerModelingRules.xif [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+| alter client_address_ipv4= arrayindex(regextract(_raw_log ,"client:\s(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"), 0)
+| alter client_address_ipv6 = arrayindex(regextract(_raw_log ,"client\:\s(\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+)") ,0)
+| alter client_address  = coalesce(client_address_ipv4 , client_address_ipv6)
+// end extract client
+| alter Server =arrayindex(regextract(_raw_log ,"server\:([^\,]+)"),0)
+| alter Request =arrayindex(regextract(_raw_log ,"request\:([^\,]+)"),0)
+| alter Host =arrayindex(regextract(_raw_log ,"host\:\s\"?([^\"?\:]+)"),0)
+| alter Referrer =arrayindex(regextract(_raw_log ,"referrer\:\s([^\s]+) "),0)
+| alter destination_port =arrayindex(regextract(_raw_log ,"host\:\s\"?[^\"?\:]+\:(\d+)"),0)
+| alter XDM.Audit.event_timestamp = parse_timestamp("%F %X", timestamp)
```
Comment (guytamir10) [inline Packs/NGINXWebServer/ModelingRules/NGINXWebServerModelingRules.xif:42]: need to change the parse string to:
%Y/%m/%d %X

**Review by evisochek — APPROVED** [review]

**Pull request merged**